tests/components/wled/test_light.py

```diff
@@ -0,0 +1,174 @@
+    assert state.attributes.get(ATTR_PLAYLIST) is None
+    assert state.attributes.get(ATTR_PRESET) is None
+    assert state.attributes.get(ATTR_SPEED) == 32
+    assert state.state == STATE_ON
+
+    entry = entity_registry.async_get("light.wled_light")
+    assert entry
+    assert entry.unique_id == "aabbccddeeff_0"
+
+    # Second segment of the strip
+    state = hass.states.get("light.wled_light_1")
+    assert state
+    assert state.attributes.get(ATTR_BRIGHTNESS) == 127
+    assert state.attributes.get(ATTR_EFFECT) == "Blink"
+    assert state.attributes.get(ATTR_HS_COLOR) == (148.941, 100.0)
+    assert state.attributes.get(ATTR_ICON) == "mdi:led-strip-variant"
+    assert state.attributes.get(ATTR_INTENSITY) == 64
+    assert state.attributes.get(ATTR_PALETTE) == "Random Cycle"
+    assert state.attributes.get(ATTR_PLAYLIST) is None
+    assert state.attributes.get(ATTR_PRESET) is None
+    assert state.attributes.get(ATTR_SPEED) == 16
+    assert state.state == STATE_ON
+
+    entry = entity_registry.async_get("light.wled_light_1")
+    assert entry
+    assert entry.unique_id == "aabbccddeeff_1"
+
+
+async def test_switch_change_state(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the change of state of the WLED switches."""
+    await init_integration(hass, aioclient_mock)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("light.wled_light")
+    assert state.state == STATE_ON
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: "light.wled_light"},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("light.wled_light")
+    assert state.state == STATE_OFF
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_ON,
+        {
+            ATTR_BRIGHTNESS: 42,
+            ATTR_EFFECT: "Chase",
+            ATTR_ENTITY_ID: "light.wled_light",
+            ATTR_RGB_COLOR: [255, 0, 0],
+            ATTR_TRANSITION: 5,
+        },
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("light.wled_light")
+    assert state.state == STATE_ON
+    assert state.attributes.get(ATTR_BRIGHTNESS) == 42
+    assert state.attributes.get(ATTR_EFFECT) == "Chase"
+    assert state.attributes.get(ATTR_HS_COLOR) == (0.0, 100.0)
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: "light.wled_light", ATTR_COLOR_TEMP: 400},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("light.wled_light")
+    assert state.state == STATE_ON
+    assert state.attributes.get(ATTR_HS_COLOR) == (28.874, 72.522)
+
+    # Test with RGBW strip
+    entity = hass.data[LIGHT_DOMAIN].get_entity("light.wled_light")
+    entity._rgbw = True  # pylint: disable=protected-access
```

Comment:
Patch `wled.device.info.leds.rgbw` instead. We should not interact with any entity interfaces in the tests. Avoiding that makes the tests more robust.

Reply:
Hmmm... maybe it is even better to just add another fixture specifically for an RGBW light, avoiding patching and workarounds all together.

Thanks 👍

Reply:
We usually prefer patching libraries instead of using response fixtures and mock rest clients, since we're not interested in testing the library, just testing our code and how we interface with the library. Usually this will mean less code in the tests.

Reply:
I partially agree. By throwing in the library itself, it ensures compatibility with the library. If the library changes behavior test would fail on a library version bump. IMHO, that is a value worth adding, and if it is just additional lines of code, a fair price to pay.

Reply:
Speccing the mock/patch with the library instance guards, to some degree, against library changes.

Reply:
I think the existing client mocks and fixture is pretty clean so far, so I don't think we need to change that. If it's just a small extra fixture needed that's ok.

Reply:
Yeah, I went that route since it allows me to improve coverage by reusing that fixture elsewhere as well.

Will push that later, lunch break is over 😉